### PR TITLE
kernel plugin: kernel targets depending on debarch

### PR DESCRIPTION
**Pull request by sergiusens**
This allows using the same snapcraft.yaml to build kernels
for different arches as they have different image targets.

LP: #1604801

Signed-off-by: Sergio Schvezov <sergio.schvezov@ubuntu.com>

### CODE REVIEW

**Comment by sergiusens** [comment]
retest this please

**Comment by sergiusens** [comment]
@leannogasawara any updates on testing for this?

**Comment by leannogasawara** [comment]
On Wed, Aug 3, 2016 at 6:21 PM, Sergio Schvezov <notifications@github.com>
wrote:

> @leannogasawara <https://github.com/leannogasawara> any updates on
> testing for this?
>

Yep, Paolo tested this last week:

<goog_1295843>
https://bugs.launchpad.net/snapcraft/+bug/1604801/comments/3

"With this patch i was able to snap, from the same kernel tree, kernels for
amd64, arm64 and armhf - looks good."

Thanks,
Leann

**Comment by sergiusens** [comment]
thanks @leannogasawara :-)

File: snapcraft/plugins/kernel.py [hunk 1]
```diff
@@ -23,8 +23,8 @@
 The following kernel specific options are provided by this plugin:
 
     - kernel-image-target:
-      (string; default: bzImage)
-      the kernel image make target to build; maps to make target.
+      (yaml object; default: bzImage)
```
Comment (elopio) [inline snapcraft/plugins/kernel.py]: Shouldn't this be: a string or a yaml object?
I think the help should explain that you can pass only one target, and how.

File: snapcraft/plugins/kernel.py [hunk 2]
```diff
@@ -157,6 +153,23 @@ def enable_cross_compilation(self):
             self.project.kernel_arch))
         self.make_cmd.append('CROSS_COMPILE={}'.format(
             self.project.cross_compiler_prefix))
+        self._set_kernel_targets()
```
Comment (elopio) [inline snapcraft/plugins/kernel.py]: I'm confused here. Why do you call it again if you already called it in the init?

Comment (sergiusens) [inline snapcraft/plugins/kernel.py]: Backwards compatibility.

Comment (sergiusens) [inline snapcraft/plugins/kernel.py]: It depends on kernel arch from the debarch

File: snapcraft/plugins/kernel.py (hunk 1 above)
Comment (sergiusens) [inline snapcraft/plugins/kernel.py]: Yeah probably so

File: snapcraft/plugins/kernel.py [hunk 3]
```diff
@@ -77,7 +82,10 @@ def schema(cls):
         schema = super().schema()
 
         schema['properties']['kernel-image-target'] = {
-            'type': 'string',
+            'oneOf': [
+                {'type': 'string'},
+                {'type': 'object'},
+            ],
```
Comment (kyrofa) [inline snapcraft/plugins/kernel.py]: I didn't know you could do this! Would have made some earlier changes easier.

**Comment by kyrofa** [comment]
I like it :+1:

**Pull request merged**